### PR TITLE
SideBar component

### PR DESCRIPTION
**Pull request by kherkatary**

### CODE REVIEW

File: src/Components/SideBar/SideBar.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "Data Structures",
+    "react",
+    "vueJs",
+    "Kotlin",
+    "pnpm",
+    "Vite",
+    "scss",
+  ];
+
+  const tagspan = tags.map((tag) => <span>{tag}</span>);
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.jsx]: Even though this is technically allowed, try to always use JSX inside the return statement (where JSX is supposed to be), as this is prone to recalculating the map.

Place this directly inside JSX.

Also, always use a key with a map. A [good reference from docs](https://reactjs.org/docs/lists-and-keys.html)

File: src/Components/SideBar/SideBar.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <div className={styles.mainBox}>
+      <div className={styles.box1}>
+        <h4>Search by tags</h4>
+        {tagspan}
+      </div>
+
+      <div className={styles.box2}>
+        <h4>Bookmarks</h4>
+        <p>What is ReactJS and how can it be used int development</p>
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.jsx]: Convert these bookmarks into links, as well as using a map to iterate over an array of bookmarks. Also, just take the first 5 of them from the array (keep more than 5 elements in the array to check).

This is done as mostly your backend data may contain all of the bookmarks.

Also side tip: For dummy content, you can use [Lorem Ipsum](https://www.lipsum.com/) which is a standard for dummy text.

File: src/Components/SideBar/SideBar.module.scss [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  justify-content: left;
+
+  width: 35%;
+  height: 80vh;
+
+  text-align: center;
+}
+
+.box1 {
+  margin: 2rem auto 2rem auto;
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.module.scss]: you can shorten this to:
margin: 2rem auto;

the first applies to top and bottom, while second for left and right.

Also, decide on a common unit. You use rem here, and vh above, and % for width as well (which is a modified version of vw which takes scrollbar into account).

Use this unit (vh-vw combo, or rem), and keep it same in as many places as possible (*not* all, but *as many as possible*)

Don't use pixel except in media query condition.

File: src/Components/SideBar/SideBar.module.scss [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  display: flex;
+  flex-wrap: wrap;
+  align-items: stretch;
+  height: auto;
+  width: 70%;
+
+  span {
+    color: grey;
+    border: 1.5px solid $border-color;
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.module.scss]: Same point about units.

File: src/Components/SideBar/SideBar.module.scss [hunk 5]
```diff
@@ -0,0 +1,94 @@
+@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;500&display=swap");
+// .............................................................
+
+$border-color: rgb(141, 141, 141, 0.2);
+
+* {
+  margin: 0;
+
+  font-family: poppins;
+}
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.module.scss]: This block and the import above is not necessary as index.scss already has these lines, and it was not imported as a module (and hence applies to all pages).

**Always read the index.scss to see variables already defined**

Also, if you want to set a font, keep the name as a variable in index.scss, like --gdsc-font-1 (which already exists, and expands to Poppins)

File: src/Components/SideBar/SideBar.module.scss [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    border-radius: 2px;
+    width: auto;
+    white-space: nowrap;
+    font-size: 0.7rem;
+    font-weight: 400;
+    padding: 1.7%;
+    margin: 1.5%;
+
+    height: fit-content;
+
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.module.scss:42]: An additional tip, you can [disable user's ability to select the text](https://www.w3schools.com/cssref/css3_pr_user-select.php), to make it more "button-like".

If you wish to implement it here, your choice.

File: src/Components/SideBar/SideBar.module.scss [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  height: auto;
+  width: 70%;
+
+  p {
+    padding: 2%;
+    text-align: left;
+    width: 100%;
+    font-size: 0.7rem;
+    font-weight: lighter;
+    margin-bottom: -10px;
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.module.scss]: Fix the unit issue in every occurence.

File: src/Pages/Home/Home.jsx [hunk 8]
```diff
@@ -10,11 +10,16 @@ const Home = () => {
...
-      {articleRespState.error && <div>Error</div>}
-      {articleRespState.response.map((item, index) => {
-        return <ArticleCard data={item} id={index} key={item?.id} />;
-      })}
+      <div className={style.articlesection}>
+        <div className={style.article}>
+          {articleRespState.loading && <Loading />}
+          {articleRespState.error && <div>Error</div>}
+          {articleRespState.response.map((item, index) => {
+            return <ArticleCard data={item} id={index} key={item?.id} />;
```
Comment (resyfer) [inline src/Pages/Home/Home.jsx]: For a single line return from an arrow function, [you don't need to write return](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#syntax)

File: src/Pages/Home/Home.module.scss [hunk 9]
```diff
@@ -4,4 +4,12 @@
   h1 {
     font-weight: bold;
   }
+  .articlesection {
```
Comment (resyfer) [inline src/Pages/Home/Home.module.scss:8]: Always add an empty line to separate two style blocks (add this change wherever necessary)

File: src/Components/SideBar/SideBar.module.scss [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  flex-direction: column;
+  justify-content: left;
+
+  width: 35%;
+  height: 80vh;
+
+  text-align: center;
+}
+
+.box1 {
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.module.scss]: Improve naming of classes. Variables should not be named box1, box2

**Review by resyfer — CHANGES_REQUESTED** [review]

File: src/Components/SideBar/SideBar.jsx [hunk 11]
```diff
@@ -37,7 +37,9 @@ const SideBar = () => {
       <div className={styles.searchTags}>
         <h4>Search by tags</h4>
         {tags.slice(0, 15).map((tag) => (
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.jsx:39]: Add a key for map. *Always* add a key for the map.

**Review by resyfer — CHANGES_REQUESTED** [review]

File: src/Components/SideBar/SideBar.jsx [hunk 12]
```diff
@@ -1,44 +1,36 @@
+import { useState } from "react";
 import styles from "./SideBar.module.scss";
-
+import sidebarData from "./sidebarData.json";
```
Comment (resyfer) [inline src/Components/SideBar/SideBar.jsx:4]: Use the fetch hook made by @JDeepD. Ask him how to use it, understand how it was written, and import the data using the hook (ie. in essence, using fetch API)

**Review by resyfer — CHANGES_REQUESTED** [review]

**Review by codadept — APPROVED** [review]

**Review by codadept — APPROVED** [review]
